cpp-client/deephaven/dhcore/include/public/deephaven/dhcore/interop/interop_util.h

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+ */
+#pragma once
+#include <cstdint>
+#include <exception>
+#include <iostream>
+#include <locale>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace deephaven::dhcore::interop {
+/**
+ * This class simply wraps a pointer. It is meant to mirror a similar struct on the C# side, namely
+ *
+ * [StructLayout(LayoutKind.Sequential)]
+ * public struct NativePtr<T> {
+ *   public IntPtr ptr;
+ * }
+ *
+ * The purpose of all of this is to enable a little more type checking on the C# side.
+ * Rather than just dealing with IntPtr on the C# side (which is effectively equivalent to
+ * void* and has just as little typechecking, we use NativePtr<Foo> struct instead.
+ * This prevents us from making mistakes like passing a NativePtr<Foo> when we meant
+ * NativePtr<Bar>. Our protocol is that .NET passes us a NativePtr<Foo> by value, so
+ * in our C API, we receive a NativePtr<Foo> by value.
+ *
+ * There is a little cheat I use because the pointers aren't fully typechecked.
```

Comment:
Suggests replacing "I use" with "we use"; similarly thereafter on the other instances of "I".

Reply:
done